### PR TITLE
[cleanup] reference to C++ and Python client

### LINKED ISSUE #17916: Cleanup or migrate all reference to pulsar-client-cpp and the python client
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Motivation

Subtasks of https://github.com/apache/pulsar/issues/17724.

### Solution

_No response_

### Alternatives

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by tisonkun**
This closes #17916.

### Documentation

- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [x] `doc` 
- [ ] `doc-complete`

### Matching PR in forked repository

PR in forked repository: https://github.com/tisonkun/pulsar/pull/4

### CODE REVIEW

**Comment by tisonkun** [comment]
cc @merlimat @Anonymitaet @BewareMyPower

**Review by lhotari — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -1529,16 +1529,6 @@ flexible messaging model and an intuitive client API.</description>
...
-                <exclude>pulsar-client-cpp/lib/checksum/gf2.hpp</exclude>
-                <exclude>pulsar-client-cpp/lib/checksum/crc32c_sse42.cc</exclude>
-                <exclude>pulsar-client-cpp/lib/checksum/crc32c_sse42.h</exclude>
-                <exclude>pulsar-client-cpp/lib/checksum/crc32c_sw.cc</exclude>
-                <exclude>pulsar-client-cpp/lib/lz4/lz4.*</exclude>
-                <exclude>pulsar-client-cpp/lib/PulsarApi.pb.*</exclude>
-                <exclude>pulsar-client-cpp/CMakeFiles/**</exclude>
-                <exclude>pulsar-client-cpp/**/Makefile</exclude>
-                <exclude>pulsar-client-cpp/**/cmake_install.cmake</exclude>
                 <exclude>**/CMakeFiles/**</exclude>
```
Comment (merlimat) [inline pom.xml:1532]: This line too should be removed

File: pom.xml [hunk 2]
```diff
@@ -1695,18 +1676,7 @@ flexible messaging model and an intuitive client API.</description>
...
             <exclude>**/cmake_install.cmake</exclude>
-            <exclude>pulsar-client-cpp/**/Makefile</exclude>
-            <exclude>pulsar-client-cpp/tests/main</exclude>
-            <exclude>pulsar-client-cpp/examples/SampleAsyncProducer</exclude>
-            <exclude>pulsar-client-cpp/examples/SampleConsumer</exclude>
-            <exclude>pulsar-client-cpp/examples/SampleConsumerListener</exclude>
-            <exclude>pulsar-client-cpp/examples/SampleProducer</exclude>
-            <exclude>pulsar-client-cpp/perf/perfProducer</exclude>
-            <exclude>pulsar-client-cpp/perf/perfConsumer</exclude>
             <exclude>**/python/dist/**</exclude>
```
Comment (merlimat) [inline pom.xml:1679]: `python/dist` should go away too

File: pom.xml [hunk 3]
```diff
@@ -1695,18 +1676,7 @@ flexible messaging model and an intuitive client API.</description>
...
-            <exclude>**/pulsar-client-cpp/pkg/rpm/RPMS/**</exclude>
-            <exclude>**/pulsar-client-cpp/pkg/rpm/SOURCES/**</exclude>
-            <exclude>**/pulsar-client-cpp/pkg/deb/BUILD/**</exclude>
             <exclude>**/python/wheelhouse/**</exclude>
             <exclude>**/python/MANIFEST</exclude>
             <exclude>**/*.egg-info/**</exclude>
```
Comment (merlimat) [inline pom.xml:1682]: These 3 lines are also python specific

**Comment by tisonkun** [comment]
@merlimat thank you! Comments addressed.

**Review by merlimat — APPROVED** [review]

**Comment by tisonkun** [comment]
@merlimat @BewareMyPower CI failed on https://github.com/apache/pulsar/issues/17921

I don't know whether it's related to this patch: I can pass the test locally, and it seems logically unrelated. But also I don't ever see it before. Perhaps rerun once more to see.

**Comment by tisonkun** [comment]
It seems rerun does not work.

ping @merlimat @lhotari @BewareMyPower

**Comment by BewareMyPower** [comment]
@tisonkun The tests passed in my local env, I'll rerun again.

**Comment by Anonymitaet** [comment]
@tisonkun thank you! I've updated other related occurrences https://github.com/apache/pulsar-site/pull/245

**Pull request merged**